contents/docker/Docker.md

```diff
@@ -0,0 +1,165 @@
+<frontmatter>
+  title: DevOps
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Docker
+
+Authors: [Rahul Rajesh](https://github.com/rrtheonlyone)
+
+In today’s age, web applications are becoming more complex. Web applications of recent times are starting to move from a monolithic architecture to one that is based off of microservices. 
+
+<box type="info">
+A Microservices architecture is when you split your application into a set of smaller, interconnected services that each have a specific role (as opposed to one big application). Read up more about it <a href="https://articles.microservices.com/monolithic-vs-microservices-architecture-5c4848858f59">here.</a>
+</box>
+
+With the increasing complexity of these applications, deployment starts to become a concern. Instead of running one service, some applications will require you to run a few services at the same time. There also needs to be configurations setup for each of these services. Applications like Docker help to make deployment a seamless process. This article provides an introduction to Docker along with the advantages that it is able to provide.
+
+### What is Docker?
+
+<img src="Docker_Logo.png" alt="Docker Logo" width="45%">
+
+<sub>_Figure 1. Docker Logo_ [(source)](https://www.theverge.com/circuitbreaker/2018/5/25/17386716/docker-kubernetes-containers-explained)</sub>
+
+Docker is a platform that is used to develop, deploy and run applications inside “containers”.
+
+> A container is a standard unit of software that packages up code and all its dependencies so the application runs quickly and reliably from one computing environment to another. <sub>--[Docker Website](https://www.docker.com/resources/what-container)</sub>
+
+The idea of containers is not so different from Virtual Machines (VM). Before the rise of Docker, people used VMs. A VM is an emulation of a real computer that is used to isolate an application and its dependencies into a self-contained unit that can run anywhere.  For example, you could use a VM to run an application on a Linux system even though you are running Windows.
+
+However, a container posses some advantages over a VM. As per the [Docker Documentation](https://docs.docker.com/get-started/), a container runs natively on Linux and shares the kernel of the host machine with other containers. It runs a discrete process, taking no more memory than any other executable, making it lightweight.
+
+By contrast, a virtual machine (VM) runs a full-blown “guest” operating system with virtual access to host resources through a hypervisor. In general, VMs provide an environment with more resources than most applications need. The diagram below gives a depiction of this:
+
+<img src="Docker_Architecture.png" alt="Docker Architecture" width="100%">
```

Comment:
It's best to put the diagram just _above_ the text that refers to it.

Reply:
Done